// @vuemap/vue-amap/es/index.mjs
import installer from './defaults.mjs';
export { default } from './defaults.mjs';
import './services/index.mjs';
import './packages/index.mjs';
export { makeInstaller, registerMixin } from '@vuemap/vue-amap-util';
export { initAMapApiLoader, lazyAMapApiLoaderInstance } from './services/injected-amap-api-instance.mjs';
export { ElAmap } from './packages/amap/index.mjs';
export { ElAmapControlControlBar } from './packages/control/ControlBar/index.mjs';
export { ElAmapControlHawkEye } from './packages/control/HawkEye/index.mjs';
export { ElAmapControlMapType } from './packages/control/MapType/index.mjs';
export { ElAmapControlScale } from './packages/control/Scale/index.mjs';
export { ElAmapControlToolBar } from './packages/control/ToolBar/index.mjs';
export { ElAmapSearchBox } from './packages/control/SearchBox/index.mjs';
export { ElAmapInfoWindow } from './packages/infoWindow/InfoWindow/index.mjs';
export { ElAmapLayerCanvas } from './packages/layer/data/Canvas/index.mjs';
export { ElAmapLayerCustom } from './packages/layer/data/Custom/index.mjs';
export { ElAmapLayerFlexible } from './packages/layer/data/Flexible/index.mjs';
export { ElAmapLayerGlCustom } from './packages/layer/data/GLCustom/index.mjs';
export { ElAmapLayerHeatMap } from './packages/layer/data/HeatMap/index.mjs';
export { ElAmapLayerImage } from './packages/layer/data/Image/index.mjs';
export { ElAmapLayerLabels } from './packages/layer/data/Labels/index.mjs';
export { ElAmapLayerVector } from './packages/layer/data/Vector/index.mjs';
export { ElAmapLayerDistrictCluster } from './packages/layer/data/DistrictCluster/index.mjs';
export { ElAmapLayerBuildings } from './packages/layer/official/Buildings/index.mjs';
export { ElAmapLayerDefault } from './packages/layer/official/DefaultLayer/index.mjs';
export { ElAmapLayerDistrict } from './packages/layer/official/DistrictLayer/index.mjs';
export { ElAmapLayerIndoorMap } from './packages/layer/official/IndoorMap/index.mjs';
export { ElAmapLayerRoadNet } from './packages/layer/official/RoadNet/index.mjs';
export { ElAmapLayerSatellite } from './packages/layer/official/Satellite/index.mjs';
export { ElAmapLayerTile } from './packages/layer/official/TileLayer/index.mjs';
export { ElAmapLayerTraffic } from './packages/layer/official/Traffic/index.mjs';
export { ElAmapLayerMapboxVectorTile } from './packages/layer/standard/MapboxVectorTileLayer/index.mjs';
export { ElAmapLayerWms } from './packages/layer/standard/WMS/index.mjs';
export { ElAmapLayerWmts } from './packages/layer/standard/WMTS/index.mjs';
export { ElAmapElasticMarker } from './packages/marker/ElasticMarker/index.mjs';
export { ElAmapLabelMarker } from './packages/marker/LabelMarker/index.mjs';
export { ElAmapMarker } from './packages/marker/Marker/index.mjs';
export { ElAmapMarkerCluster } from './packages/marker/MarkerCluster/index.mjs';
export { ElAmapMassMarks } from './packages/marker/MassMarks/index.mjs';
export { ElAmapText } from './packages/marker/Text/index.mjs';
export { ElAmapBezierCurve } from './packages/vector/BezierCurve/index.mjs';
export { ElAmapCircle } from './packages/vector/Circle/index.mjs';
export { ElAmapEllipse } from './packages/vector/Ellipse/index.mjs';
export { ElAmapGeojson } from './packages/vector/GeoJSON/index.mjs';
export { ElAmapPolygon } from './packages/vector/Polygon/index.mjs';
export { ElAmapPolyline } from './packages/vector/Polyline/index.mjs';
export { ElAmapRectangle } from './packages/vector/Rectangle/index.mjs';
export { ElAmapLayerTiles3d } from './packages/layer/data/Tiles3D/index.mjs';
export { ElAmapControlGeolocation } from './packages/control/Geolocation/index.mjs';
export { ElAmapCircleMarker } from './packages/marker/CircleMarker/index.mjs';
export { ElAmapLayerVideo } from './packages/layer/data/Video/index.mjs';
export { ElAmapMouseTool } from './packages/util/MouseTool/index.mjs';

const install = installer.install;

export { install };
