// @vuemap/vue-amap/lib/packages/control/MapType/MapType.vue_vue_type_script_lang.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var vueAmapUtil = require('@vuemap/vue-amap-util');

var script = vue.defineComponent({
  name: "ElAmapControlMapType",
  mixins: [vueAmapUtil.registerMixin],
  props: {
    defaultType: {
      type: Number
    },
    showTraffic: {
      type: Boolean,
      default: false
    },
    showRoad: {
      type: Boolean,
      default: false
    }
  },
  data() {
    return {};
  },
  methods: {
    __initComponent(options) {
      return new Promise((resolve) => {
        this.$parentComponent.plugin(["AMap.MapType"], () => {
          this.$amapComponent = new AMap.MapType(options);
          this.$parentComponent.addControl(this.$amapComponent);
          resolve();
        });
      });
    },
    destroyComponent() {
      if (this.$amapComponent && this.$parentComponent) {
        if (!this.parentInstance.isDestroy) {
          this.$parentComponent.removeControl(this.$amapComponent);
        }
        this.$amapComponent = null;
        this.$parentComponent = null;
      }
    }
  },
  render() {
    return null;
  }
});

exports["default"] = script;
