// @vuemap/vue-amap/es/packages/layer/official/Traffic/Traffic.vue_vue_type_script_lang.mjs
import { defineComponent } from 'vue';
import { registerMixin } from '@vuemap/vue-amap-util';

var script = defineComponent({
  name: "ElAmapLayerTraffic",
  mixins: [registerMixin],
  props: {
    autoRefresh: {
      type: Boolean,
      defult: true
    },
    interval: {
      type: Number
    },
    zooms: {
      type: Array
    },
    opacity: {
      type: Number
    },
    tileSize: {
      type: Number
    }
  },
  data() {
    return {};
  },
  methods: {
    __initComponent(options) {
      this.$amapComponent = new AMap.TileLayer.Traffic(options);
      this.$parentComponent.add(this.$amapComponent);
    },
    stopFresh() {
      if (this.$amapComponent) {
        this.$amapComponent.stopFresh();
      }
    },
    destroyComponent() {
      if (!this.parentInstance.isDestroy) {
        this.$parentComponent.remove(this.$amapComponent);
      }
      this.$amapComponent = null;
      this.$parentComponent = null;
    }
  },
  render() {
    return null;
  }
});

export { script as default };
