// @vuemap/vue-amap/lib/packages/layer/official/Traffic/Traffic.vue_vue_type_script_lang.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var vueAmapUtil = require('@vuemap/vue-amap-util');

var script = vue.defineComponent({
  name: "ElAmapLayerTraffic",
  mixins: [vueAmapUtil.registerMixin],
  props: {
    autoRefresh: {
      type: Boolean,
      defult: true
    },
    interval: {
      type: Number
    },
    zooms: {
      type: Array
    },
    opacity: {
      type: Number
    },
    tileSize: {
      type: Number
    }
  },
  data() {
    return {};
  },
  methods: {
    __initComponent(options) {
      this.$amapComponent = new AMap.TileLayer.Traffic(options);
      this.$parentComponent.add(this.$amapComponent);
    },
    stopFresh() {
      if (this.$amapComponent) {
        this.$amapComponent.stopFresh();
      }
    },
    destroyComponent() {
      if (!this.parentInstance.isDestroy) {
        this.$parentComponent.remove(this.$amapComponent);
      }
      this.$amapComponent = null;
      this.$parentComponent = null;
    }
  },
  render() {
    return null;
  }
});

exports["default"] = script;
