// @vuemap/vue-amap/lib/packages/marker/LabelMarker/LabelMarker.vue_vue_type_script_lang.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var vueAmapUtil = require('@vuemap/vue-amap-util');

var script = vue.defineComponent({
  name: "ElAmapLabelMarker",
  mixins: [vueAmapUtil.registerMixin],
  props: {
    name: {
      type: String
    },
    position: {
      type: [Array, Object],
      required: true
    },
    zooms: {
      type: Array
    },
    opacity: {
      type: Number
    },
    rank: {
      type: Number
    },
    icon: {
      type: Object
    },
    text: {
      type: Object
    },
    extData: null
  },
  data() {
    return {
      converters: {}
    };
  },
  methods: {
    __initComponent(options) {
      this.$amapComponent = new AMap.LabelMarker(options);
      this.$parentComponent.add(this.$amapComponent);
    },
    destroyComponent() {
      if (!this.parentInstance.isDestroy) {
        this.$parentComponent.remove(this.$amapComponent);
      }
      this.$amapComponent = null;
      this.$parentComponent = null;
    }
  },
  render() {
    return null;
  }
});

exports["default"] = script;
