// lodash-es@4.17.21/math.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var add = require('./add.js');
var ceil = require('./ceil.js');
var divide = require('./divide.js');
var floor = require('./floor.js');
var max = require('./max.js');
var maxBy = require('./maxBy.js');
var mean = require('./mean.js');
var meanBy = require('./meanBy.js');
var min = require('./min.js');
var minBy = require('./minBy.js');
var multiply = require('./multiply.js');
var round = require('./round.js');
var subtract = require('./subtract.js');
var sum = require('./sum.js');
var sumBy = require('./sumBy.js');
var math_default = require('./math.default.js');



exports.add = add["default"];
exports.ceil = ceil["default"];
exports.divide = divide["default"];
exports.floor = floor["default"];
exports.max = max["default"];
exports.maxBy = maxBy["default"];
exports.mean = mean["default"];
exports.meanBy = meanBy["default"];
exports.min = min["default"];
exports.minBy = minBy["default"];
exports.multiply = multiply["default"];
exports.round = round["default"];
exports.subtract = subtract["default"];
exports.sum = sum["default"];
exports.sumBy = sumBy["default"];
exports["default"] = math_default["default"];
