// @vuemap/vue-amap/lib/packages/control/Geolocation/Geolocation.vue_vue_type_script_lang.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var vueAmapUtil = require('@vuemap/vue-amap-util');

var script = vue.defineComponent({
  name: "ElAmapControlGeolocation",
  mixins: [vueAmapUtil.registerMixin],
  props: {
    position: {
      type: String
    },
    offset: {
      type: Array
    },
    borderColor: {
      type: String
    },
    borderRadius: {
      type: String
    },
    buttonSize: {
      type: String
    },
    convert: {
      type: Boolean,
      default: true
    },
    enableHighAccuracy: {
      type: Boolean,
      default: false
    },
    timeout: {
      type: Number
    },
    maximumAge: {
      type: Number
    },
    showButton: {
      type: Boolean,
      default: true
    },
    showCircle: {
      type: Boolean,
      default: true
    },
    showMarker: {
      type: Boolean,
      default: true
    },
    markerOptions: {
      type: Object
    },
    circleOptions: {
      type: Object
    },
    panToLocation: {
      type: Boolean,
      default: true
    },
    zoomToAccuracy: {
      type: Boolean,
      default: true
    },
    GeoLocationFirst: {
      type: Boolean,
      default: true
    },
    noIpLocate: {
      type: Number
    },
    noGeoLocation: {
      type: Number
    },
    useNative: {
      type: Boolean,
      default: false
    },
    getCityWhenFail: {
      type: Boolean,
      default: false
    },
    needAddress: {
      type: Boolean,
      default: false
    },
    extensions: {
      type: String,
      validator(value) {
        return ["base", "all"].indexOf(value) !== -1;
      }
    }
  },
  emits: ["complete"],
  data() {
    return {};
  },
  methods: {
    __initComponent(options) {
      return new Promise((resolve) => {
        this.$parentComponent.plugin(["AMap.Geolocation"], () => {
          this.$amapComponent = new AMap.Geolocation(options);
          this.$parentComponent.addControl(this.$amapComponent);
          this.$amapComponent.on("complete", (e) => {
            this.$emit("complete", e);
          });
          resolve();
        });
      });
    },
    destroyComponent() {
      if (this.$amapComponent && this.$parentComponent) {
        if (!this.parentInstance.isDestroy) {
          this.$parentComponent.removeControl(this.$amapComponent);
        }
        this.$amapComponent = null;
        this.$parentComponent = null;
      }
    }
  },
  render() {
    return null;
  }
});

exports["default"] = script;
