// @vuemap/vue-amap/es/packages/layer/standard/MapboxVectorTileLayer/MapboxVectorTileLayer.vue_vue_type_script_lang.mjs
import { defineComponent } from 'vue';
import { registerMixin } from '@vuemap/vue-amap-util';

var script = defineComponent({
  name: "ElAmapLayerMapboxVectorTile",
  mixins: [registerMixin],
  props: {
    url: {
      type: String
    },
    zooms: {
      type: Array
    },
    dataZooms: {
      type: Array
    },
    opacity: {
      type: Number
    },
    styles: {
      type: Object
    }
  },
  data() {
    return {};
  },
  methods: {
    __initComponent(options) {
      return new Promise((resolve) => {
        AMap.plugin(["AMap.MapboxVectorTileLayer"], () => {
          this.$amapComponent = new AMap.MapboxVectorTileLayer(options);
          this.$parentComponent.addLayer(this.$amapComponent);
          resolve();
        });
      });
    },
    destroyComponent() {
      if (!this.parentInstance.isDestroy) {
        this.$parentComponent.removeLayer(this.$amapComponent);
      }
      this.$amapComponent = null;
      this.$parentComponent = null;
    }
  },
  render() {
    return null;
  }
});

export { script as default };
