// @vuemap/vue-amap/lib/packages/layer/official/Buildings/Buildings.vue_vue_type_script_lang.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var vueAmapUtil = require('@vuemap/vue-amap-util');

var script = vue.defineComponent({
  name: "ElAmapLayerBuildings",
  mixins: [vueAmapUtil.registerMixin],
  props: {
    wallColor: {
      type: [String, Array]
    },
    roofColor: {
      type: [String, Array]
    },
    heightFactor: {
      type: Number
    },
    styleOpts: {
      type: Object
    },
    zooms: {
      type: Array
    },
    opacity: {
      type: Number
    }
  },
  data() {
    return {};
  },
  methods: {
    __initComponent(options) {
      this.$amapComponent = new AMap.Buildings(options);
      this.$parentComponent.add(this.$amapComponent);
      if (this.styleOpts) {
        this.__styleOpts(this.styleOpts);
      }
    },
    destroyComponent() {
      this.$amapComponent.destroy();
      if (!this.parentInstance.isDestroy) {
        this.$parentComponent.remove(this.$amapComponent);
      }
      this.$amapComponent = null;
      this.$parentComponent = null;
    },
    __styleOpts(value) {
      this.$amapComponent.setStyle(value);
    }
  },
  render() {
    return null;
  }
});

exports["default"] = script;
