// @vuemap/vue-amap/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defaults = require('./defaults.js');
require('./services/index.js');
require('./packages/index.js');
var vueAmapUtil = require('@vuemap/vue-amap-util');
var injectedAmapApiInstance = require('./services/injected-amap-api-instance.js');
var index = require('./packages/amap/index.js');
var index$1 = require('./packages/control/ControlBar/index.js');
var index$2 = require('./packages/control/HawkEye/index.js');
var index$3 = require('./packages/control/MapType/index.js');
var index$4 = require('./packages/control/Scale/index.js');
var index$5 = require('./packages/control/ToolBar/index.js');
var index$6 = require('./packages/control/SearchBox/index.js');
var index$7 = require('./packages/infoWindow/InfoWindow/index.js');
var index$8 = require('./packages/layer/data/Canvas/index.js');
var index$9 = require('./packages/layer/data/Custom/index.js');
var index$a = require('./packages/layer/data/Flexible/index.js');
var index$b = require('./packages/layer/data/GLCustom/index.js');
var index$c = require('./packages/layer/data/HeatMap/index.js');
var index$d = require('./packages/layer/data/Image/index.js');
var index$e = require('./packages/layer/data/Labels/index.js');
var index$f = require('./packages/layer/data/Vector/index.js');
var index$g = require('./packages/layer/data/DistrictCluster/index.js');
var index$h = require('./packages/layer/official/Buildings/index.js');
var index$i = require('./packages/layer/official/DefaultLayer/index.js');
var index$j = require('./packages/layer/official/DistrictLayer/index.js');
var index$k = require('./packages/layer/official/IndoorMap/index.js');
var index$l = require('./packages/layer/official/RoadNet/index.js');
var index$m = require('./packages/layer/official/Satellite/index.js');
var index$n = require('./packages/layer/official/TileLayer/index.js');
var index$o = require('./packages/layer/official/Traffic/index.js');
var index$p = require('./packages/layer/standard/MapboxVectorTileLayer/index.js');
var index$q = require('./packages/layer/standard/WMS/index.js');
var index$r = require('./packages/layer/standard/WMTS/index.js');
var index$s = require('./packages/marker/ElasticMarker/index.js');
var index$t = require('./packages/marker/LabelMarker/index.js');
var index$u = require('./packages/marker/Marker/index.js');
var index$v = require('./packages/marker/MarkerCluster/index.js');
var index$w = require('./packages/marker/MassMarks/index.js');
var index$x = require('./packages/marker/Text/index.js');
var index$y = require('./packages/vector/BezierCurve/index.js');
var index$z = require('./packages/vector/Circle/index.js');
var index$A = require('./packages/vector/Ellipse/index.js');
var index$B = require('./packages/vector/GeoJSON/index.js');
var index$C = require('./packages/vector/Polygon/index.js');
var index$D = require('./packages/vector/Polyline/index.js');
var index$E = require('./packages/vector/Rectangle/index.js');
var index$F = require('./packages/layer/data/Tiles3D/index.js');
var index$G = require('./packages/control/Geolocation/index.js');
var index$H = require('./packages/marker/CircleMarker/index.js');
var index$I = require('./packages/layer/data/Video/index.js');
var index$J = require('./packages/util/MouseTool/index.js');

const install = defaults["default"].install;

exports["default"] = defaults["default"];
Object.defineProperty(exports, 'makeInstaller', {
	enumerable: true,
	get: function () { return vueAmapUtil.makeInstaller; }
});
Object.defineProperty(exports, 'registerMixin', {
	enumerable: true,
	get: function () { return vueAmapUtil.registerMixin; }
});
exports.initAMapApiLoader = injectedAmapApiInstance.initAMapApiLoader;
Object.defineProperty(exports, 'lazyAMapApiLoaderInstance', {
	enumerable: true,
	get: function () { return injectedAmapApiInstance.lazyAMapApiLoaderInstance; }
});
exports.ElAmap = index.ElAmap;
exports.ElAmapControlControlBar = index$1.ElAmapControlControlBar;
exports.ElAmapControlHawkEye = index$2.ElAmapControlHawkEye;
exports.ElAmapControlMapType = index$3.ElAmapControlMapType;
exports.ElAmapControlScale = index$4.ElAmapControlScale;
exports.ElAmapControlToolBar = index$5.ElAmapControlToolBar;
exports.ElAmapSearchBox = index$6.ElAmapSearchBox;
exports.ElAmapInfoWindow = index$7.ElAmapInfoWindow;
exports.ElAmapLayerCanvas = index$8.ElAmapLayerCanvas;
exports.ElAmapLayerCustom = index$9.ElAmapLayerCustom;
exports.ElAmapLayerFlexible = index$a.ElAmapLayerFlexible;
exports.ElAmapLayerGlCustom = index$b.ElAmapLayerGlCustom;
exports.ElAmapLayerHeatMap = index$c.ElAmapLayerHeatMap;
exports.ElAmapLayerImage = index$d.ElAmapLayerImage;
exports.ElAmapLayerLabels = index$e.ElAmapLayerLabels;
exports.ElAmapLayerVector = index$f.ElAmapLayerVector;
exports.ElAmapLayerDistrictCluster = index$g.ElAmapLayerDistrictCluster;
exports.ElAmapLayerBuildings = index$h.ElAmapLayerBuildings;
exports.ElAmapLayerDefault = index$i.ElAmapLayerDefault;
exports.ElAmapLayerDistrict = index$j.ElAmapLayerDistrict;
exports.ElAmapLayerIndoorMap = index$k.ElAmapLayerIndoorMap;
exports.ElAmapLayerRoadNet = index$l.ElAmapLayerRoadNet;
exports.ElAmapLayerSatellite = index$m.ElAmapLayerSatellite;
exports.ElAmapLayerTile = index$n.ElAmapLayerTile;
exports.ElAmapLayerTraffic = index$o.ElAmapLayerTraffic;
exports.ElAmapLayerMapboxVectorTile = index$p.ElAmapLayerMapboxVectorTile;
exports.ElAmapLayerWms = index$q.ElAmapLayerWms;
exports.ElAmapLayerWmts = index$r.ElAmapLayerWmts;
exports.ElAmapElasticMarker = index$s.ElAmapElasticMarker;
exports.ElAmapLabelMarker = index$t.ElAmapLabelMarker;
exports.ElAmapMarker = index$u.ElAmapMarker;
exports.ElAmapMarkerCluster = index$v.ElAmapMarkerCluster;
exports.ElAmapMassMarks = index$w.ElAmapMassMarks;
exports.ElAmapText = index$x.ElAmapText;
exports.ElAmapBezierCurve = index$y.ElAmapBezierCurve;
exports.ElAmapCircle = index$z.ElAmapCircle;
exports.ElAmapEllipse = index$A.ElAmapEllipse;
exports.ElAmapGeojson = index$B.ElAmapGeojson;
exports.ElAmapPolygon = index$C.ElAmapPolygon;
exports.ElAmapPolyline = index$D.ElAmapPolyline;
exports.ElAmapRectangle = index$E.ElAmapRectangle;
exports.ElAmapLayerTiles3d = index$F.ElAmapLayerTiles3d;
exports.ElAmapControlGeolocation = index$G.ElAmapControlGeolocation;
exports.ElAmapCircleMarker = index$H.ElAmapCircleMarker;
exports.ElAmapLayerVideo = index$I.ElAmapLayerVideo;
exports.ElAmapMouseTool = index$J.ElAmapMouseTool;
exports.install = install;
