// @vuemap/vue-amap/es/packages/marker/MarkerCluster/MarkerCluster.vue_vue_type_script_lang.mjs
import { defineComponent } from 'vue';
import { registerMixin } from '@vuemap/vue-amap-util';

var script = defineComponent({
  name: "ElAmapMarkerCluster",
  mixins: [registerMixin],
  props: {
    points: {
      type: Array,
      required: true
    },
    gridSize: {
      type: Number
    },
    maxZoom: {
      type: Number
    },
    averageCenter: {
      type: Boolean,
      default: true
    },
    clusterByZoomChange: {
      type: Boolean,
      default: false
    },
    styles: {
      type: Array
    },
    renderClusterMarker: {
      type: Function
    },
    renderMarker: {
      type: Function
    }
  },
  data() {
    return {
      converters: {}
    };
  },
  methods: {
    __initComponent(options) {
      return new Promise((resolve) => {
        AMap.plugin(["AMap.MarkerCluster"], () => {
          const points = options.points;
          delete options.points;
          this.$amapComponent = new AMap.MarkerCluster(this.$parentComponent, points, options);
          resolve();
        });
      });
    },
    destroyComponent() {
      this.$amapComponent.setMap(null);
      this.$amapComponent = null;
      this.$parentComponent = null;
    },
    __points(value) {
      if (this.$amapComponent) {
        this.$amapComponent.setData(value);
      }
    }
  },
  render() {
    return null;
  }
});

export { script as default };
