// @vuemap/vue-amap/lib/packages/layer/standard/MapboxVectorTileLayer/MapboxVectorTileLayer.vue_vue_type_script_lang.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var vueAmapUtil = require('@vuemap/vue-amap-util');

var script = vue.defineComponent({
  name: "ElAmapLayerMapboxVectorTile",
  mixins: [vueAmapUtil.registerMixin],
  props: {
    url: {
      type: String
    },
    zooms: {
      type: Array
    },
    dataZooms: {
      type: Array
    },
    opacity: {
      type: Number
    },
    styles: {
      type: Object
    }
  },
  data() {
    return {};
  },
  methods: {
    __initComponent(options) {
      return new Promise((resolve) => {
        AMap.plugin(["AMap.MapboxVectorTileLayer"], () => {
          this.$amapComponent = new AMap.MapboxVectorTileLayer(options);
          this.$parentComponent.addLayer(this.$amapComponent);
          resolve();
        });
      });
    },
    destroyComponent() {
      if (!this.parentInstance.isDestroy) {
        this.$parentComponent.removeLayer(this.$amapComponent);
      }
      this.$amapComponent = null;
      this.$parentComponent = null;
    }
  },
  render() {
    return null;
  }
});

exports["default"] = script;
