// @vuemap/vue-amap/lib/mixins/editor-component.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var vueAmapUtil = require('@vuemap/vue-amap-util');

var editorMixin = vue.defineComponent({
  props: {
    editable: {
      type: Boolean,
      default: void 0
    },
    editOptions: {
      type: Object
    }
  },
  data() {
    return {};
  },
  methods: {
    setEditorEvents() {
      if (!this.$amapComponent.editor)
        return;
      const filters = ["addnode", "adjust", "removenode", "end", "move", "add"];
      const filterSet = {};
      Object.keys(this.$attrs).forEach((key) => {
        if (vueAmapUtil.eventReg.test(key)) {
          const eventKey = vueAmapUtil.convertEventToLowerCase(key);
          if (filters.indexOf(eventKey) !== -1)
            filterSet[eventKey] = this.$attrs[key];
        }
      });
      Object.keys(filterSet).forEach((key) => {
        vueAmapUtil.eventHelper.addListener(this.$amapComponent.editor, key, filterSet[key]);
      });
    }
  }
});

exports["default"] = editorMixin;
